sdk/storage/Azure.Storage.DataMovement/src/JobPlanModels/JobPartPlanFile.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+
+namespace Azure.Storage.DataMovement
+{
+    internal class JobPartPlanFile : IDisposable
+    {
+        /// <summary>
+        /// Save the associated file name within a struct. This will contain
+        /// our transfer id, job part id, verison etc.
+        /// </summary>
+        public JobPartPlanFileName FileName { get; set; }
+
+        /// <summary>
+        /// Used when opening the MemoryMappedFileReference
+        /// </summary>
+        public string MapName { get; internal set; }
+
+        /// <summary>
+        /// The associated file on disk. When the last process has finished working
+        /// with the file, the data is saved to the file on the disk.
+        /// </summary>
+        public string FilePath { get => FileName.ToString(); }
+
+        /// <summary>
+        /// Lock for the memory mapped file to allow only one writer.
+        /// </summary>
+        public readonly SemaphoreSlim WriteLock;
+
+        private JobPartPlanFile()
+        {
+            WriteLock = new SemaphoreSlim(1);
+        }
+
+        public static async Task<JobPartPlanFile> CreateJobPartPlanFileAsync(
+            string checkpointerPath,
+            string id,
+            int jobPart,
+            Stream headerStream)
+        {
+            Argument.AssertNotNullOrEmpty(checkpointerPath, nameof(checkpointerPath));
+            Argument.AssertNotNullOrEmpty(id, nameof(id));
+            Argument.AssertNotNull(jobPart, nameof(jobPart));
+            Argument.AssertNotNull(headerStream, nameof(headerStream));
+
+            string mapName = string.Concat(id, jobPart);
+            JobPartPlanFileName fileName = new JobPartPlanFileName(checkpointerPath: checkpointerPath, id: id, jobPartNumber: jobPart);
```

Comment:
It's exists to maintain the file name structure since it's formatted in specific way. You're right we could just store it as a string, but it was just to stay organized, having the checks live in the same place and parsing to live in the same place.